XMonad/Prompt/DotDesktopParser.hs

```diff
@@ -0,0 +1,171 @@
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE LambdaCase #-}
+module XMonad.Prompt.DotDesktopParser
+    ( runDotDesktopParser
+    ) where
+
+import Data.Maybe ( catMaybes )
+import Control.Monad ( MonadPlus(..) )
+import Control.Applicative ( Alternative(..) )
+import qualified Data.Map as MAP
+
+newtype Parser a = Parser { parse :: String -> [(a,String)] }
```

Comment:
There already is quite a capable parser in `base` called [ReadP](https://hackage.haskell.org/package/base-4.16.0.0/docs/Text-ParserCombinators-ReadP.html).  Some modules, like [X.P.OrgMode](https://hackage.haskell.org/package/xmonad-contrib-0.17.0/docs/src/XMonad.Prompt.OrgMode.html#line-355), already use that and so it would be nice if this could as well.  

I wouldn't be opposed to a general X.U.Parser though, since the commutative nature of `ReadP` ironically makes some things harder (see e.g. [here](https://hackage.haskell.org/package/xmonad-contrib-0.17.0/docs/src/XMonad.Prompt.OrgMode.html#endBy1))

Reply:
I like this idea!  I think will take me a while to re-implement as I am unfamiliar with ReadR but I will start on it.

Reply:
re-implemented this using ReadP.  While it is working for me, I probably need to test this more.